src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -563,6 +568,73 @@ private static IEnumerable<string> DiscoverBindingTypes(IEnumerable<FunctionMeta
             return bindingTypes;
         }
 
+        // Validate that for any precompiled assembly, all functions have the same configuration precedence.
+        // Also load any extensions from the precompiled directory.
+        private void VerifyPrecompiled(Collection<FunctionMetadata> functions)
+        {
+            HashSet<string> extensionFolders = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            Dictionary<string, bool> mapAssemblySettings = new Dictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
+            foreach (var function in functions)
+            {
+                var scriptFile = function.ScriptFile;
+                if (scriptFile.EndsWith(".dll", StringComparison.OrdinalIgnoreCase))
+                {
+                    bool isDirect = function.IsDirect;
+                    bool prevIsDirect;
+                    if (mapAssemblySettings.TryGetValue(scriptFile, out prevIsDirect))
+                    {
+                        if (prevIsDirect != isDirect)
+                        {
+                            // Error. All entries pointing to the same dll must have the same value for IsDirect
+                            string msg = string.Format(CultureInfo.InvariantCulture, "Configuration error: All functions in {0} must have the same config precedence,",
+                                scriptFile);
+                            TraceWriter.Info(msg);
+                            _startupLogger?.LogInformation(msg);
+
+                            // Hard failure
+                            functions.Clear();
```

Comment:
Throwing seems like the correct thing to do. The other endpoints don't require a running host. This would, however, initiate a new load loop.